gateway/src/main/java/discord4j/gateway/intent/IntentSet.java

```diff
@@ -0,0 +1,256 @@
+package discord4j.gateway.intent;
+
+import java.util.*;
+
+/**
+ * An <b>immutable</b>, specialized {@code Set<Intent>}.
+ *
+ * <p>
+ * This is a <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/doc-files/ValueBased.html">value-based</a>
+ * class; use of identity-sensitive operations (including reference equality
+ * ({@code ==}), identity hash code, or synchronization) on instances of
+ * {@code IntentSet} may have unpredictable results and should be avoided.
+ * The {@code equals} method should be used for comparisons.
+ *
+ * @see <a href="https://discordapp.com/developers/docs/topics/intents">Discord Intents</a>
+ */
+public final class IntentSet extends AbstractSet<Intent> {
```

Comment:
Does anyone have any ideas for de-duplication with PermissionSet? Java might not be enough of a friend to do it nicely, but I'd like to ask for ideas on it now rather than later.

Reply:
I tried that first, but I ran into an issue I wasn't able to resolve - but that was a few weeks ago, I am not sure which one it was. Probably some kind of inheritance issue...